### PR TITLE
Declare IPVS-based kube-proxy GA

### LINKED ISSUE #58443: Declare IPVS-based kube-proxy GA
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

/sig network

/area ipvs

**What happened**:

As discussed in sig-network meeting, we decided to target IPVS-based kube-proxy GA in 1.10.

**What you expected to happen**:

Declare IPVS-based kube-proxy GA and remove feature gateway.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

As discussed in sig-network meeting, we decided to target IPVS-based kube-proxy GA in 1.11.

This PR declares IPVS-based kube-proxy GA and remove feature gateway.

**Which issue(s) this PR fixes**:

Fixes: #58443

issue in feature repo: https://github.com/kubernetes/features/issues/265

**Special notes for your reviewer**:

**Release note**:

```release-note
Declare IPVS-based kube-proxy GA
```

@quinton-hoole

### CODE REVIEW

**Comment by m1093782566** [comment]
Slightly ping @thockin ~

I remember we have discussed it in sig network meeting and had already reached a agreement?

**Comment by rramkumar1** [comment]
@m1093782566 PR #56732 needs to be resolved in some fashion before ipvs can move to GA.

**Comment by rramkumar1** [comment]
@m1093782566 Issue #59427 needs to be resolved as well

**Comment by thockin** [comment]
Assign back to me when all child PRs are in?

**Comment by m1093782566** [comment]
Sure. My team member @Lion-Wei @stewart-yu commit to fix the child PRs.

**Comment by squeed** [comment]
I've added a new networking tests that checks for working service-hairpin in #59929. It would be good to see that merged and passing with ipvs.

(Service hairpin, i.e. Pod A -> Service -> Pod A, was previously broken in ipvs)

**Comment by m1093782566** [comment]
the hairpin issue is fixed in IPVS mode now. anyway, ipvs will run that once the test case get merged.

File: cmd/kube-proxy/app/server_others.go [hunk 1]
```diff
@@ -280,18 +278,14 @@ func getProxyMode(proxyMode string, iptver iptables.IPTablesVersioner, khandle i
 		return proxyModeUserspace
 	}
 
-	if len(proxyMode) > 0 && proxyMode == proxyModeIPTables {
+	if proxyMode == proxyModeIPTables {
```
Comment (thockin) [inline cmd/kube-proxy/app/server_others.go]: can we change this to a switch ?  Including the userspace above

File: cmd/kube-proxy/app/server_others_test.go [hunk 2]
```diff
@@ -88,146 +87,47 @@ func Test_getProxyMode(t *testing.T) {
 			kernelCompat:    true,
 			expected:        proxyModeIPTables,
 		},
-		{ // specify ipvs, feature gateway disabled, iptables version ok, kernel is compatible
```
Comment (thockin) [inline cmd/kube-proxy/app/server_others_test.go]: Why does this remove so many tests?

File: pkg/features/kube_features.go [hunk 3]
```diff
@@ -210,12 +210,6 @@ const (
 	// Enable resource limits priority function
 	ResourceLimitsPriorityFunction utilfeature.Feature = "ResourceLimitsPriorityFunction"
 
-	// owner: @m1093782566
-	// beta: v1.9
-	//
-	// Implement IPVS-based in-cluster service load balancing
-	SupportIPVSProxyMode utilfeature.Feature = "SupportIPVSProxyMode"
```
Comment (thockin) [inline pkg/features/kube_features.go]: We can't remove this until after it is deprecated, so this needs a release-note "action required".  See "Rule #5b: CLI elements of admin-facing components" in https://kubernetes.io/docs/reference/using-api/deprecation-policy/

**Review by thockin — COMMENTED** [review]
Get that KEP in, too.

File: pkg/features/kube_features.go (hunk 3 above)
Comment (m1093782566) [reply pkg/features/kube_features.go]: Thanks for your response, Tim.

IIUC, we should keep the feature gateway for IPVS and just flip the `beta` to `GA` and then deprecate this after 1+ release cycles?

**Comment by m1093782566** [comment]
> Get that KEP in, too.

Okay, I will cooperate with @rramkumar1 to get the KEP in.

@thockin ALL comments are fixed now. PTAL. Can we push this forward in this(v1.11) release cycle?

**Comment by m1093782566** [comment]
/kind feature

priority/important-soon

/sig network

**Pull request merged**